### PR TITLE
Feature/#494

### PR DESCRIPTION
**Pull request by JakubVanek**
- implements new LCD init/flush - needs testing on both Stretch and Jessie
  - jessie might not require expensive flush anymore, the underlying raster hopefully stores the data in the correct format
- clean up and optimize a bit the properties system

### CODE REVIEW

File: src/main/java/ev3dev/hardware/EV3DevPlatforms.java [hunk 1]
```diff
@@ -28,7 +30,65 @@ public EV3DevPlatform getPlatform() {
     }
 
     public String getMotorPort(final Port port) {
-        return instance.getMotorPort(port);
+
+        //TODO Pending to review in detail
+        //return instance.getMotorPort(port);
+
```
Comment (JakubVanek) [inline src/main/java/ev3dev/hardware/EV3DevPlatforms.java]: This is probably a wrong place to do such a revert. If you want to use the original implementation, switch it in EV3DevPlatformsImpl. Changing it there defeats the caching.

File: src/test/java/fake_ev3dev/BaseElement.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 public abstract class BaseElement {
 
     private static final String JAVA_IO_TEMPDIR = System.getProperty("java.io.tmpdir");
-    public static final String EV3DEV_FAKE_SYSTEM_PATH = Paths.get(JAVA_IO_TEMPDIR, "ev3dev_fake_system").toString();
+    public static final String EV3DEV_FAKE_SYSTEM_PATH = JAVA_IO_TEMPDIR + "/" + "ev3dev_fake_system";
```
Comment (JakubVanek) [inline src/test/java/fake_ev3dev/BaseElement.java]: This does not work on Linux.:
```java
JAVA_IO_TEMPDIR = "/tmp"; // and that means
EV3DEV_FAKE_SYSTEM_PATH = "/tmpev3dev_fake_system";
```
and programs cannot write to /.

File: src/main/java/ev3dev/hardware/EV3DevPlatforms.java [hunk 3]
```diff
@@ -28,7 +30,65 @@ public EV3DevPlatform getPlatform() {
...
+            } else if (port.equals(MotorPort.D)) {
+                return ev3DevProperties.getProperty("pistorms.motor.port.d");
+            }
+
+        }
+
+        throw new RuntimeException("Bad port used");
     }
 
     public String getSensorPort(final Port port) {
```
Comment (JakubVanek) [inline src/main/java/ev3dev/hardware/EV3DevPlatforms.java]: If you want to postpone the giant if-ladder changes, there is also one missing.

File: src/test/java/fake_ev3dev/BaseElement.java (hunk 2 above)
Comment (JakubVanek) [reply src/test/java/fake_ev3dev/BaseElement.java]: Hmmm, I have not noticed the slash, now it should work.

**Comment by jabrena** [comment]
We will change it.
https://github.com/ev3dev-lang-java/ev3dev-lang-java/issues/532

**Pull request closed without merging**